sphinx/util/inspect.py

```diff
@@ -244,8 +244,15 @@ def object_description(object):
         except TypeError:
             pass  # Cannot sort dict keys, fall back to generic repr
         else:
-            items = ("%r: %r" % (key, object[key]) for key in sorted_keys)
+            items = ("%s: %s" %
+                     (object_description(key), object_description(object[key]))
+                     for key in sorted_keys)
             return "{%s}" % ", ".join(items)
+    if isinstance(object, set):
+        # Sort set contents
+        template = "{%s}" if PY3 else "set([%s])"
+        return template % ", ".join(object_description(x)
+                                    for x in sorted(object))
```

Comment:
This will be failed if members are not sortable:
```
>>> sorted({1, 2, 'str'})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '<' not supported between instances of 'str' and 'int'
```
```
>>> sorted({1, 2, object()})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '<' not supported between instances of 'object' and 'int'
```

Reply:
Good shout. Have updated with a testcase.